src/main/java/com/avispl/symphony/dal/avdevices/power/surgex/defenderseries/SurgeXDefenderCommunicator.java

```diff
@@ -0,0 +1,529 @@
+		});
+
+		return properties;
+	}
+
+	/**
+	 * Generates network setting properties from the networkSettings.
+	 *
+	 * @return A map of network property names and their corresponding values,
+	 * or an empty map if the network settings are not available.
+	 */
+	private Map<String, String> getNetworkSettingProperties() {
+		if (Objects.isNull(this.networkSetting)) {
+			return Collections.emptyMap();
+		}
+		return this.generateProperties(
+				NetworkSettingProperty.values(),
+				Constant.NETWORK_GROUP,
+				property -> Util.mapToNetworkSettingProperty(this.networkSetting, property)
+		);
+	}
+
+	/**
+	 * Generates outlet-related properties from the currentStatus.
+	 *
+	 * @return A map of outlet property names and values grouped by outlet index,
+	 * or an empty map if no outlets are found.
+	 */
+	private Map<String, String> getOutletProperties() {
+		List<Outlet> deviceOutlets = Util.getOutlets(this.currentStatus);
+		if (CollectionUtils.isEmpty(deviceOutlets)) {
+			return Collections.emptyMap();
+		}
+		Map<String, String> properties = new HashMap<>();
+		for (int i = 0; i < deviceOutlets.size(); i++) {
+			Outlet outlet = deviceOutlets.get(i);
+			properties.putAll(this.generateProperties(
+					OutletProperty.values(),
+					String.format(Constant.GROUP_FORMAT, Constant.OUTLET_GROUP, i + 1),
+					property -> Util.mapToOutletProperty(outlet, property)
+			));
+		}
+
+		return properties;
+	}
+
+	/**
+	 * Generates properties for outlet groups from the currentStatus.
+	 *
+	 * @return A map of outlet group property names and values grouped by group index,
+	 * or an empty map if no outlet groups are found.
+	 */
+	private Map<String, String> getOutletGroupProperties() {
+		List<Group> deviceOutletGroups = Util.getOutletGroups(this.currentStatus);
+		if (CollectionUtils.isEmpty(deviceOutletGroups)) {
+			return Collections.emptyMap();
+		}
+		Map<String, String> properties = new HashMap<>();
+		for (int i = 0; i < deviceOutletGroups.size(); i++) {
+			Group group = deviceOutletGroups.get(i);
+			properties.putAll(this.generateProperties(
+					OutletGroupProperty.values(),
+					String.format(Constant.GROUP_FORMAT, Constant.GROUP_OUTLET_GROUP, i + 1),
+					property -> Util.mapToOutletGroupProperty(group, property)
+			));
+		}
+
+		return properties;
+	}
+
+	/**
+	 * Generates controllable properties (e.g. toggle, reboot) for each outlet.
+	 *
+	 * @return A list of controllable properties for outlets,
+	 * or an empty list if no outlets are found.
+	 */
+	private List<AdvancedControllableProperty> getOutletControllers() {
+		List<Outlet> deviceOutlets = Util.getOutlets(this.currentStatus);
+		if (CollectionUtils.isEmpty(deviceOutlets)) {
+			return Collections.emptyList();
+		}
+		List<AdvancedControllableProperty> properties = new ArrayList<>();
+		for (int i = 0; i < deviceOutlets.size(); i++) {
+			Outlet outlet = deviceOutlets.get(i);
+			if (Util.isRebootingComponent(outlet.getState())) {
+				continue;
```

Comment:
please add log entry here, so it's clear what's going on during debugging/troubleshooting